storage/txcache/evictionScoreComputer.go

```diff
@@ -0,0 +1,145 @@
+package txcache
+
+import "math"
+
+type evictionScoreComputer struct {
+	senders          []*txListForSender
+	scores           []float64
+	scoresAsPercents []int64
+
+	maxGas         int64
+	minGas         int64
+	maxSize        int64
+	minSize        int64
+	maxOrderNumber int64
+	minOrderNumber int64
+	maxTxCount     int64
+	minTxCount     int64
+
+	gasRange         int64
+	sizeRange        int64
+	orderNumberRange int64
+	txCountRange     int64
+}
+
+func newEvictionScoreComputer(senders []*txListForSender) *evictionScoreComputer {
```

Comment:
I think this is pretty time consuming when the tx is at a limit, and there will be a high noise of eviction and putting new transactions. When full size eviction is reached, we might evict 50% of the cache.

Reply:
We don't have anymore a score computer that iterates over all senders. Now we compute the score when a new transaction arrives for a given sender. See `OnScoreChange()`.